frame/fee-market/src/lib.rs

```diff
@@ -241,53 +245,73 @@ pub mod pallet {
 				T::RingCurrency::free_balance(&who) >= new_collateral,
 				<Error<T>>::InsufficientBalance
 			);
-			ensure!(
-				new_collateral > Self::get_relayer(&who).collateral,
-				<Error<T>>::OnlyIncreaseLockedCollateralAllowed
-			);
-
-			Self::update_collateral(&who, new_collateral);
-			Self::deposit_event(Event::<T>::UpdateLockedCollateral(who, new_collateral));
-			Ok(().into())
-		}
 
-		/// Cancel enrolled relayer
-		#[pallet::weight(<T as Config>::WeightInfo::cancel_enrollment())]
-		#[transactional]
-		pub fn cancel_enrollment(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
-			let who = ensure_signed(origin)?;
-			ensure!(Self::is_enrolled(&who), <Error<T>>::NotEnrolled);
+			let relayer = Self::relayer(&who);
 			ensure!(
-				<Relayers<T>>::get().len() > T::AssignedRelayersNumber::get() as usize,
-				<Error<T>>::TooFewEnrolledRelayers
+				new_collateral > relayer.collateral,
+				<Error<T>>::OnlyIncCollateralAllowed
 			);
```

Comment:
In the new model, it should be ok to reduce collateral as soon as it does not affect the locked collateral in Orders.

CollateralForOrder * count_of_orders_in_delivering, or the sum of the collateral in delivering orders.

Reply:
Right, we can remove the original `OnlyIncCollateralAllowed` limit.

Reply:
Now it supports decrease locked collateral. please review again.